app/account/page.jsx

```diff
@@ -0,0 +1,109 @@
+'use client'
+import * as React from 'react';
+import PropTypes from 'prop-types';
+import Tabs from '@mui/material/Tabs';
+import Tab from '@mui/material/Tab';
+import Typography from '@mui/material/Typography';
+import Box from '@mui/material/Box';
+import { useEffect, useState } from 'react'
+import { useUserContext } from '../context/usercontext'
+import { useRouter } from 'next/navigation'
+import Profile from '../components/profile'
+
+function CustomTabPanel(props) {
+  const { children, value, index, ...other } = props;
+
+  
+
+  return (
+    <div
+      role="tabpanel"
+      hidden={value !== index}
+      id={`simple-tabpanel-${index}`}
+      aria-labelledby={`simple-tab-${index}`}
+      {...other}
+    >
+      {value === index && (
+        <Box sx={{ p: 3 }}>
+          <Typography>{children}</Typography>
+        </Box>
+      )}
+    </div>
+  );
+}
+
+CustomTabPanel.propTypes = {
+  children: PropTypes.node,
+  index: PropTypes.number.isRequired,
+  value: PropTypes.number.isRequired,
+};
+
+function a11yProps(index) {
+  return {
+    id: `simple-tab-${index}`,
+    'aria-controls': `simple-tabpanel-${index}`,
+  };
+}
+
+export default function BasicTabs() {
+    const [value, setValue] = React.useState(0);
+    
+    const {user, setUser, userData, setUserData} = useUserContext();
+    const [dataLoaded, setDataLoaded] = useState(false);
+    let prevUser = null;
+    const router = useRouter();
+
+    useEffect(()=>{
+        if (user===null) {
+        try {
+            prevUser = JSON.parse(sessionStorage.getItem('firebase:authUser:AIzaSyCoGURJeUWdIylWkAEDYEpOqY6YnAaJYy0:[DEFAULT]'))
+            setUser(prevUser)
+        } catch {}
+        }
+    })
+    
+
+    useEffect(()=>{
+
+        user===null && prevUser===null && router.push('/registration')
+    },[])
+
+    useEffect(()=>{
+       if (user === null && prevUser===null) {
+          router.push('/signin') 
+        } else if (!userData.first && user!==null) {
+
+            async function fetchUser(action, collection, document) {
```

Comment:
best practice here is not to pollute your code with internally scoped functions. This could get confusing, makes it more difficult to read your code, and also decreases the testability of this specific function. You should declare these outside of other functions.